redbot/cogs/mutes/mutes.py

```diff
@@ -344,81 +355,65 @@ async def on_guild_channel_update(
-            if member.id in self._server_mutes[guild.id]:
-                until = self._server_mutes[guild.id][member.id]["until"]
-                del self._server_mutes[guild.id][member.id]
-        for channel in guild.channels:
-            if channel.id in self._channel_mutes:
-                if member.id in self._channel_mutes[channel.id]:
-                    del self._channel_mutes[channel.id][member.id]
-        mute_role = await self.config.guild(guild).mute_role()
-        if not mute_role:
-            return
-        if mute_role in [r.id for r in member.roles]:
-            async with self.config.guild(guild).removed_users() as removed_users:
-                removed_users[str(member.id)] = int(until) if until else None
+                    log.debug("creating case")
+                    await modlog.create_case(
+                        self.bot,
+                        after.guild,
+                        datetime.now(timezone.utc),
+                        "cunmute",
+                        user,
+                        None,
+                        _("Manually removed channel overwrites"),
+                        until=None,
+                        channel=after,
+                    )
+                    log.debug("created case")
+            if to_del:
+                for u_id in to_del:
+                    del self._channel_mutes[after.id][u_id]
+                await self.config.channel(after).muted_users.set(self._channel_mutes[after.id])
 
     @commands.Cog.listener()
     async def on_member_join(self, member: discord.Member):
         guild = member.guild
         mute_role = await self.config.guild(guild).mute_role()
         if not mute_role:
             return
-        async with self.config.guild(guild).removed_users() as removed_users:
-            if str(member.id) in removed_users:
-                until_ts = removed_users[str(member.id)]
-                until = datetime.fromtimestamp(until_ts, tz=timezone.utc) if until_ts else None
-                # datetime is required to utilize the mutes method
-                if until and until < datetime.now(tz=timezone.utc):
-                    return
-                removed_users.pop(str(member.id))
+        if guild.id in self._server_mutes:
+            if member.id in self._server_mutes[guild.id]:
                 role = guild.get_role(mute_role)
                 if not role:
                     return
+                until = self._server_mutes[guild.id][member.id]["until"]
                 await self.mute_user(
                     guild, guild.me, member, until, _("Previously muted in this server.")
                 )
 
     @commands.group()
     @commands.guild_only()
-    @checks.mod_or_permissions(manage_roles=True)
     async def muteset(self, ctx: commands.Context):
         """Mute settings."""
         pass
 
+    @muteset.command(name="notification")
+    @checks.mod_or_permissions(manage_messages=True)
```

Comment:
*I think* it would make more sense for this setting to be available only to people with admin role or those having manage channels permission as I don't think that most administrators would want moderators to have access to this kind of setting and it is a setting that affects all moderators and visibility of these notifications.
```suggestion
    @checks.admin_or_permissions(manage_channels=True)
```